cognite/neat/_rules/transformers/_converters.py

```diff
@@ -232,56 +234,144 @@ def __init__(
+            new_model.metadata.version = cast(str, self.new_model_id.version)
+
+        if reference_model_id in COGNITE_MODELS:
+            new_model.reference = reference_model
+
+        if self.mode == "solution":
+            return self._to_solution(new_model, reference_model_id)
+
+        elif self.mode == "extension":
+            return self._to_extension(new_model, reference_model_id)
+
+    def _to_solution(self, rules: DMSRules, reference_model_id: DataModelId) -> JustRules[DMSRules]:
+        """For creation of solution data model / rules specifically for mapping over existing containers."""
+
+        # Dropping containers since this model is only for mapping on
+        # existing containers from the reference model.
+        rules.containers = None
+
+        # We are setting any implements to None since model is expected to be used
+        # for consumption only of existing containers.
+        for view in rules.views:
+            view.implements = None
+
+        # If reference model on which we are mapping one of Cognite Data Models
+        if reference_model_id in COGNITE_MODELS:
+            # Remove CognitePrefixes.
+            for prop in rules.properties:
+                prop.view = self._remove_cognite_affix(prop.view)
+                prop.class_ = self._remove_cognite_affix(prop.class_)
                 if isinstance(prop.value_type, ViewEntity):
-                    prop.value_type = self._remove_cognite_prefix(prop.value_type)
-            for view in new_model.views:
-                view.view = self._remove_cognite_prefix(view.view)
-                view.class_ = self._remove_cognite_prefix(view.class_)
+                    prop.value_type = self._remove_cognite_affix(prop.value_type)
+            for view in rules.views:
+                view.view = self._remove_cognite_affix(view.view)
+                view.class_ = self._remove_cognite_affix(view.class_)
+
+        return JustRules(rules)
+
+    def _to_extension(self, rules: DMSRules, reference_model_id: DataModelId) -> JustRules[DMSRules]:
+        new_views: list[DMSView] = []
+        new_containers: list[DMSContainer] = []
+        new_properties: list[DMSProperty] = []
+
+        for definition in rules.views:
+            view_entity = self._remove_cognite_affix(definition.view)
+
+            view_entity.version = cast(str, self.new_model_id.version)
+            view_entity.prefix = self.new_model_id.space
+            container_entity = ContainerEntity(space=view_entity.prefix, externalId=view_entity.external_id)
+            class_entity = ClassEntity(prefix=view_entity.prefix, suffix=view_entity.suffix)
+
+            view = DMSView(
+                view=view_entity,
+                implements=[definition.view],
+                in_model=True,
+                class_=class_entity,
+                name=definition.name,
+            )
 
-        return JustRules(new_model)
+            container = DMSContainer(
```

Comment:
Should an extension have containers? I don't think a pure extension should have containers at all, that is what you add when you actually do an extension

Reply:
I see this is connected to the property below. I don't understand this part.

Reply:
Discussion with Fredrik now and I understand why you do this. Essentially a patch that bypasses the need for any fancy filters on the view. I see the value, going for this should have consequences how we set default filtering, as well as building solution models.